x-pack/plugins/security_solution/public/common/components/charts/donutchart.tsx

```diff
@@ -0,0 +1,152 @@
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText, useEuiTheme } from '@elastic/eui';
+import React, { useMemo } from 'react';
+
+import {
+  Chart,
+  Datum,
+  Partition,
+  Settings,
+  PartitionLayout,
+  defaultPartitionValueFormatter,
+  NodeColorAccessor,
+  PartialTheme,
+} from '@elastic/charts';
+import styled from 'styled-components';
+import { useTheme } from './common';
+import { DraggableLegend } from './draggable_legend';
+import { LegendItem } from './draggable_legend_item';
+import { DonutChartEmpty } from './donutchart_empty';
+
+export const NO_LEGEND_DATA: LegendItem[] = [];
+
+const donutTheme: PartialTheme = {
+  chartMargins: { top: 0, bottom: 0, left: 0, right: 0 },
+  partition: {
+    idealFontSizeJump: 1.1,
+    outerSizeRatio: 1,
+    emptySizeRatio: 0.8,
+    circlePadding: 4,
+  },
+};
+
+interface DonutChartData {
+  key: string;
+  value: number;
+  group?: string;
+  label?: string;
+}
+
+export type FillColor = string | NodeColorAccessor;
+export interface DonutChartProps {
+  data: DonutChartData[] | null | undefined;
+  fillColor: FillColor;
+  height?: number;
+  label: string;
+  legendItems?: LegendItem[] | null | undefined;
+  link?: string | null;
+  title: React.ReactElement | string | number | null;
+  totalCount: number | null | undefined;
+}
+
+/* Make this position absolute in order to overlap the text onto the donut */
+const DonutTextWrapper = styled(EuiFlexGroup)`
+  top: 34%;
+  width: 100%;
+  max-width: 77px;
+  position: absolute;
+  z-index: 1;
+`;
+
+const StyledEuiFlexItem = styled(EuiFlexItem)`
+  position: relative;
+  align-items: center;
+`;
+
+export const DonutChart = ({
+  data,
+  fillColor,
+  height = 90,
+  label,
+  legendItems,
+  link,
+  title,
+  totalCount,
+}: DonutChartProps) => {
+  const theme = useTheme();
+  const { euiTheme } = useEuiTheme();
+  const emptyLabelStyle = useMemo(
+    () => ({
+      color: euiTheme.colors.disabled,
+    }),
+    [euiTheme.colors.disabled]
+  );
+  return (
+    <EuiFlexGroup
+      alignItems="center"
+      justifyContent="center"
+      responsive={false}
+      gutterSize="l"
+      data-test-subj="donut-chart"
+    >
+      <StyledEuiFlexItem grow={false}>
+        <DonutTextWrapper
+          direction="column"
+          gutterSize="none"
+          alignItems="center"
+          justifyContent="center"
+        >
+          <EuiFlexItem>{title}</EuiFlexItem>
+          <EuiFlexItem className="eui-textTruncate">
+            {!data && (
+              <EuiText className="eui-textTruncate" size="s" style={emptyLabelStyle}>
+                {label}
+              </EuiText>
+            )}
+            {data && !link && (
+              <EuiText className="eui-textTruncate" size="s">
+                {label}
+              </EuiText>
+            )}
+
+            {data && link && <EuiLink className="eui-textTruncate">{label}</EuiLink>}
```

Comment:
I see the `link` is not passed to the `EuiLink`. 
Even though we are never passing the links for the chart title currently, this is code that is never used. We don't know when will we have to add links here, or what kind of action will be needed, probably we will also need to add a callback to open a confirmation tooltip or to dispatch some filter action....

Since we still don't know how we are going to do it, it is better to not add unused code, so we remove complexity from the implementation and testing. 

We could simply do this by now:
```suggestion
            {data ? (
              <EuiText className="eui-textTruncate" size="s">
                {label}
              </EuiText>
            ) : (
              <EuiText className="eui-textTruncate" size="s" style={emptyLabelStyle}>
                {label}
              </EuiText>
            )}
```